pkg/solana/state_cache.go

```diff
@@ -23,16 +23,24 @@ type StateCache struct {
 	*client.Cache[State]
 }
 
-func NewStateCache(stateID solana.PublicKey, chainID string, cfg config.Config, reader client.Reader, lggr logger.Logger) *StateCache {
+type GetAccountReader func() (client.AccountReader, error)
+
+func NewStateCache(stateID solana.PublicKey, chainID string, cfg config.Config, reader GetReader, lggr logger.Logger) *StateCache {
```

Comment:
please double check the naming across your PR - it should be `getReader` not `reader`